server/artifacts/artifact_server.go

```diff
@@ -186,41 +182,28 @@ func (a *ArtifactServer) returnArtifact(ctx context.Context, w http.ResponseWrit
 	if err != nil {
 		return err
 	}
-	tmp, err := ioutil.TempFile("/tmp", "artifact")
-	if err != nil {
-		return err
-	}
-	tmpPath := tmp.Name()
-	defer func() { _ = os.Remove(tmpPath) }()
-
-	err = driver.Load(art, tmpPath)
-	if err != nil {
-		return err
-	}
 
-	file, err := os.Open(filepath.Clean(tmpPath))
+	stream, err := driver.OpenStream(art)
 	if err != nil {
 		return err
 	}
 
 	defer func() {
-		if err := file.Close(); err != nil {
-			log.Fatalf("Error closing file[%s]: %v", tmpPath, err)
+		if err := stream.Close(); err != nil {
+			log.Fatalf("Error closing stream[%s]: %v", stream, err)
```

Comment:
this should have been fixed ages ago - Fatal will panic and crash! We don't want that at all - instead, just ignore the error. We will never take any action.

Reply:
how about maybe just a warning? otherwise, for the drivers that are using the common code, they'll leave files around and we may be more likely to notice if there's a warning ?